crates/hotshot/src/traits/networking/web_server_network.rs

```diff
@@ -986,7 +977,17 @@ impl<TYPES: NodeType + 'static> ConnectedNetwork<Message<TYPES>, TYPES::Signatur
                     debug!("Somehow task already existed!");
                 }
 
-                // TODO ED Do we need to GC before returning?  Or will view sync task handle that?
+                // Remove all entries in the task map that are polling for a view less than or equal to `view_number - 2`.
+                let view_minus_2 = view_number.saturating_sub(2);
+                let range = task_map.range(..view_minus_2);
+                for (view, task) in range {
+                    // Cancel the old task by sending a message to it. If the task already exited we expect an error
+                    let _res = task
+                        .send(ConsensusIntentEvent::CancelPollForViewSyncCertificate(
+                            *view,
+                        ))
+                        .await;
+                }
```

Comment:
I need an expert opinion on this one. I ran tests with view sync and it looks good, but maybe there's some weird case I didn't think about. Since we weren't doing any cancelling before for view sync certs, is it probably better to just remove this?

Reply:
^^ cc @elliedavidson @bfish713

Reply:
This should be fine, view sync does have a requirement that we help others get out of view sync but that should be complete after 2 views + other nodes in our system will get future proposals to get out of view sync

Reply:
Also just double checking, you were specifically running the a webserver based test for view sync right?